### PR TITLE
#164693172: Update user's profile image

### PR DESCRIPTION
**Pull request by kleva-j**
#### What does this PR do?
- It adds the feature for users to be able to update their profile image.

#### Description of Task to be completed?
- Users should be able to update their profile image.

#### How should this be manually tested?
- Run `npm run start` to start the application.
- Make a `POST` request to `/api/v1/userprofile/image` which will enable an authenticated user to update their profile image

### What are the relevant pivotal tracker stories?
- [#164693172](https://www.pivotaltracker.com/story/show/164693172)

#### Any background context you want to provide?
- None

#### Screenshots (if appropriate)
- None

#### Questions:
- N/A

### CODE REVIEW

**Review by codinger41 — APPROVED** [review]

File: controllers/profile.js [hunk 1]
```diff
@@ -103,4 +95,31 @@ export default class ProfileController {
...
+   * @returns {object} User Profile
+   */
+  static async updateProfileImage(req, res) {
+    const { user: { id }, body: { images } } = req;
+    const image = images[0];
+    try {
+      const [, [{
+        name, email, username, bio, image: profilePic, createdAt, updatedAt
+      }]] = await User.update({ image }, { returning: true, raw: true, where: { id } });
+      return res.status(205).json({
```
Comment (youngestdj) [inline controllers/profile.js:114]: Great job Michael. Is there a reason why you're returning 205? I am of the opinion that a 201 or 200 would be appropriate here. According to https://httpstatuses.com/205, a 205 response code would imply that no additional content will be provided, and you're providing a body in your response. Kindly look into this.

File: middleware/validation.js [hunk 2]
```diff
@@ -240,3 +240,9 @@ export const validateArticleId = [
     .isUUID()
     .withMessage('Please provide a valid id for the article')
 ];
+
+export const validateImages = [
```
Comment (youngestdj) [inline middleware/validation.js:304]: Are you checking for maximum image size and that the image is actually an image?

File: controllers/profile.js (hunk 1 above)
Comment (kleva-j) [reply controllers/profile.js:114]: Thank you for this feedback. I have made use of the material you shared and I now have a better understanding of the appropriate error response to use for this request.

File: middleware/validation.js (hunk 2 above)
Comment (kleva-j) [reply middleware/validation.js:304]: Yes, I took those factors into consideration and handled it in a separate middleware.

Comment (youngestdj) [reply middleware/validation.js:304]: Great!

**Comment by 9jaswag** [comment]
@kleva-j what's the update on this?

**Comment by 9jaswag** [comment]
@kleva-j ??

**Comment by 9jaswag** [comment]
@kleva-j can I get a screen share for this?

**Review by 9jaswag — APPROVED** [review]

**Pull request merged**